### PR TITLE
Remove deprecations for Pillow 10.0.0

### PR DESCRIPTION
**Pull request by hugovk**
Remove deprecations scheduled for removal in Pillow 10.0.0:

* Tk/Tcl <= 8.4 support was deprecated in 8.2.0
  https://github.com/python-pillow/Pillow/pull/5216

* `im.category` and related `Image.NORMAL`, `Image.SEQUENCE` and `Image.CONTAINER` attributes, deprecated in 8.2.0
  https://github.com/python-pillow/Pillow/pull/5351

* `JpegImagePlugin.convert_dict_qtables`, deprecated in 8.3.0
  https://github.com/python-pillow/Pillow/pull/4989

* `ImagePalette` `size` parameter, deprecated in 8.4.0
  https://github.com/python-pillow/Pillow/pull/5641

* `ImageShow.Viewer.show_file()` `file` argument, deprecated in 9.1.0
  https://github.com/python-pillow/Pillow/pull/5959

* Constants deprecated in 9.1.0
  https://github.com/python-pillow/Pillow/pull/5954

* `FitsStubImagePlugin`, deprecated in 9.1.0
  https://github.com/python-pillow/Pillow/pull/6056

* `FreeTypeFont.getmask2()` `fill` parameter, deprecated in 9.2.0
  https://github.com/python-pillow/Pillow/pull/6220

* `PhotoImage.paste()` `box` parameter, deprecated in 9.2.0
  https://github.com/python-pillow/Pillow/pull/6178

* Qt5 support, deprecated in Pillow 9.2.0
  https://github.com/python-pillow/Pillow/pull/6237

* `Image.coerce_e`, deprecated in 9.2.0
   https://github.com/python-pillow/Pillow/pull/6254

TODO in followup:

* Font size and offset methods
  https://github.com/python-pillow/Pillow/pull/6381

* [x] Release notes

### CODE REVIEW

**Comment by hugovk** [comment]
Will remove the other deprecations here too.

**Comment by hugovk** [comment]
Question about [`6369c80` (#7059)](https://github.com/python-pillow/Pillow/pull/7059/commits/6369c80ed642b56f9560078be47386b6baf849a3):

> * `ImageShow.Viewer.show_file()` `file` argument, deprecated in 9.1.0
>   [Deprecated show_file "file" argument in favour of "path" #5959](https://github.com/python-pillow/Pillow/pull/5959)

Should we also do this?

```diff
-    def show_file(self, path=None, **options):
+    def show_file(self, path, **options):
         """
         Display given file.
         """
-        if path is None:
-            msg = "Missing required argument: 'path'"
-            raise TypeError(msg)
         os.system(self.get_command(path, **options))  # nosec
         return 1
```

**Comment by hugovk** [comment]
Question about [`4c9f8b4` (#7059)](https://github.com/python-pillow/Pillow/pull/7059/commits/4c9f8b49bd7be33f4e821b7e4a61a1efd89b6659):

> * `Image.coerce_e`, deprecated in 9.2.0
>   [Support more affine expression forms in im.point(), deprecate Image.coerce_e #6254](https://github.com/python-pillow/Pillow/pull/6254)

`class _E` has this comment:
```python
# _E(scale, offset) represents the affine transformation scale * x + offset.
# The "data" field is named for compatibility with the old implementation,
# and should be renamed once coerce_e is removed.
```

What should it be renamed to?

**Comment by radarhere** [comment]
> Question about [`6369c80` (#7059)](https://github.com/python-pillow/Pillow/pull/7059/commits/6369c80ed642b56f9560078be47386b6baf849a3):
> 
> > * `ImageShow.Viewer.show_file()` `file` argument, deprecated in 9.1.0
> >   [Deprecated show_file "file" argument in favour of "path" #5959](https://github.com/python-pillow/Pillow/pull/5959)
> 
> Should we also do this?
> 
> ```diff
> -    def show_file(self, path=None, **options):
> +    def show_file(self, path, **options):
>          """
>          Display given file.
>          """
> -        if path is None:
> -            msg = "Missing required argument: 'path'"
> -            raise TypeError(msg)
>          os.system(self.get_command(path, **options))  # nosec
>          return 1
> ```

Yes, sounds like good to me. The default value of `None` was only added in #5959 - my intention would have been just to allow both `path` and `file` to work during the deprecation period.

**Comment by radarhere** [comment]
> Question about [`4c9f8b4` (#7059)](https://github.com/python-pillow/Pillow/pull/7059/commits/4c9f8b49bd7be33f4e821b7e4a61a1efd89b6659):
> 
> > * `Image.coerce_e`, deprecated in 9.2.0
> >   [Support more affine expression forms in im.point(), deprecate Image.coerce_e #6254](https://github.com/python-pillow/Pillow/pull/6254)
> 
> `class _E` has this comment:
> 
> ```python
> # _E(scale, offset) represents the affine transformation scale * x + offset.
> # The "data" field is named for compatibility with the old implementation,
> # and should be renamed once coerce_e is removed.
> ```
> 
> What should it be renamed to?

From my reading of the comment, the user's plan was for it to be renamed "offset"

**Comment by nulano** [comment]
My reading of the comment is the following change:

```diff
-# _E(scale, offset) represents the affine transformation scale * x + offset.
-# The "data" field is named for compatibility with the old implementation,
-# and should be renamed once coerce_e is removed.
 class _E:
-    def __init__(self, scale, data):
+    def __init__(self, scale, offset):
         self.scale = scale
-        self.data = data
+        self.offset = offset
```

File: src/PIL/ImageFont.py [hunk 1]
```diff
@@ -665,7 +474,6 @@ def getmask2(
         self,
         text,
         mode="",
-        fill=_UNSPECIFIED,
```
[the remark this answers is not in the document]
Comment (hugovk) [reply src/PIL/ImageFont.py:668]: Yeah, let's discuss this in another issue or PR.

File: src/PIL/ImageDraw.py [hunk 2]
```diff
@@ -433,17 +431,7 @@ def _multiline_split(self, text):
...
-            warnings.filterwarnings("ignore", category=DeprecationWarning)
-            return (
-                self.textsize(
-                    "A",
-                    font=font,
-                    stroke_width=stroke_width,
-                )[1]
-                + spacing
-            )
+        return self.textbbox((0, 0), "A", font, stroke_width=stroke_width)[3] + spacing
```
[the remark this answers is not in the document]
Comment (hugovk) [reply src/PIL/ImageDraw.py:434]: Shall we drop these `ImageFont` removals from this PR and we can consider all this separately?

The other changes here fix a failure with the new 3.12.0a7 (https://github.com/python-pillow/Pillow/pull/7072#issuecomment-1498938569), so it'd be good to get that merged.

[the remark this answers is not in the document]
Comment (radarhere) [reply src/PIL/ImageDraw.py:434]: Sure, whatever you prefer.

[the remark this answers is not in the document]
Comment (hugovk) [reply src/PIL/ImageDraw.py]: Okay, made a copy of this branch and pushed to https://github.com/hugovk/Pillow/tree/rm-qt5a, then dropped the last two commits here (ac23f4f0569a12362c6dd3a7a538527bc01f376d, 77120622d7812d442941899364c6a04b0dfca54d), then rebased on main (to cleanup the fixups) and force-pushed.

[the remark this answers is not in the document]
Comment (hugovk) [reply src/PIL/ImageDraw.py]: @nulano Please could you make a followup for `ImageFont`? Thank you!

File: .github/workflows/test-cygwin.yml [hunk 3]
```diff
@@ -67,7 +67,7 @@ jobs:
             python3${{ matrix.python-minor-version }}-numpy
             python3${{ matrix.python-minor-version }}-sip
             python3${{ matrix.python-minor-version }}-tkinter
-            qt5-devel-tools
+            qt6-devel-tools
```
Comment (radarhere) [inline .github/workflows/test-cygwin.yml:70]:
```suggestion
```

qt6-devel-tools isn't found in Python [3.8](https://github.com/python-pillow/Pillow/actions/runs/4629282314/jobs/8189344115#step:4:74) or [3.9](https://github.com/python-pillow/Pillow/actions/runs/4629282314/jobs/8189344329#step:4:74)

Comment (nulano) [reply .github/workflows/test-cygwin.yml:70]: Looks like Cygwin doesn't provide any Qt6 packages: https://www.cygwin.com/packages/package_list.html#q

File: src/PIL/ImageDraw.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (radarhere) [reply src/PIL/ImageDraw.py]: I've created PR #7080 to help this along - feel free to suggest changes.

**Comment by radarhere** [comment]
I've created PR #7080 as the followup PR.

**Pull request merged**